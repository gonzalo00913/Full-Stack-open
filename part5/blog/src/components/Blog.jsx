import Togglable from "./Togglable";

const Blog = ({blog, handleLike, handleDeleteBlog}) => {
  const blogStyle = {
    paddingTop: 10,
    paddingLeft: 2,
    border: "solid",
    borderWidth: 1,
    marginBottom: 5,
  
  };
  return (
    <div style={blogStyle} className="blog">
      <div>
        <div className="blog">
        <span>{blog.title}</span>
        <p>{blog.author}</p>
        </div>
        <Togglable buttonLabel="view">
          <p>{blog.url}</p>
          <p>Likes:{blog.likes}</p>
          <button onClick={() => handleLike(blog)}>Like</button>
     
        </Togglable>
        <button onClick={handleDeleteBlog}>delete</button>
      </div>
    </div>
  );
};

export default Blog;
