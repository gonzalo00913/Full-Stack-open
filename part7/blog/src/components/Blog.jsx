import Togglable from "./Togglable";

const Blog = ({blog, handleLike, handleDeleteBlog}) => {
  const blogStyle = {
    paddingTop: 10,
    paddingLeft: 2,
    border: "solid",
    borderWidth: 1,
    marginBottom: 5,
  
  };
  return (
    <div style={blogStyle} className="blog">
      <div>
        <div className="blog">
        <span>{blog.title}</span>
        <p>{blog.author}</p>
        </div>
        <Togglable buttonLabel="view">
          <p className="blog-url">{blog.url}</p>
          <p className="blog-likes">Likes:{blog.likes}</p>
          <button onClick={() => handleLike(blog)}>Like</button>
     
        </Togglable>
        <button onClick={handleDeleteBlog}>delete</button>
      </div>
    </div>
  );
};

export default Blog;
