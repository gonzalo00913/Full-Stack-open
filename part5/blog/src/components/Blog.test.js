import React from "react";
import { render } from "@testing-library/react";
import Blog from "./Blog";

describe("Blog Component", () => {
  const blog = {
    title: "Ejemplo de Blog",
    author: "Autor del Blog",
    url: "https://ejemplo.com",
    likes: 10,
  };

  it("verifica que el componente Blog muestra tÃ­tulo y autor, pero no URL ni likes por defecto", () => {
    const component = render(<Blog blog={blog}/>);

  
    const blogElements = component.container.querySelectorAll(".blog");

   
    expect(blogElements.length).toBeGreaterThan(0);


    const titleElement = component.container.querySelector(".blog span");
    const authorElement = component.container.querySelector(".blog p");
    expect(titleElement).toBeDefined(); 
    expect(authorElement).toBeDefined(); 

   
    const urlElement = component.container.querySelector(".blog .url");
    const likesElement = component.container.querySelector(".blog .likes");
    expect(urlElement).toBeNull(); 
    expect(likesElement).toBeNull(); 
  });
});