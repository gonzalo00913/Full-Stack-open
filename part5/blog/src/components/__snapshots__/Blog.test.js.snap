// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Component muestra el componente Blog con estilo CSS 1`] = `
<div>
  <div
    class="blog"
    style="padding-top: 10px; padding-left: 2px; border: solid; border-width: 1px; margin-bottom: 5px;"
  >
    <div>
      <div>
        <span>
          Ejemplo de Blog
        </span>
        <p>
          Autor del Blog
        </p>
      </div>
      <div>
        <div>
          <button>
            view
          </button>
        </div>
        <div
          style="display: none;"
        >
          <p>
            https://ejemplo.com
          </p>
          <p>
            Likes:
            10
          </p>
          <button>
            Like
          </button>
          <button>
            Cancel
          </button>
        </div>
      </div>
      <button>
        delete
      </button>
    </div>
  </div>
</div>
`;
